tools/python/test/test_numpy_returns.py

```diff
@@ -24,17 +34,26 @@ def get_test_face_chips():
 def get_test_face_chip():
     rgb_img, shape = get_test_image_and_shape()
     return dlib.get_face_chip(rgb_img, shape)
- 
+
+# The tests below will be skipped for Python 2.7 as there is a pickle issue
```

Comment:
This should work in python 2.7.  Can't you pickle using an earlier pickling version, like 2? It's certainly not the case that pybind11 pickling doesn't work in python 2.7.

Reply:
If you use cPickle and a call like .dumps(variable,2) it works.  See http://pybind11.readthedocs.io/en/stable/advanced/classes.html#pickling-support